funcbench/main.go

```diff
@@ -149,82 +185,85 @@ The default is 2 hours (2h).`).Default("2h").Duration()
-		if pErr := env.PostErr("Go bench test for this pull request failed"); pErr != nil {
-			return errors.Errorf("error: %v occured while processing error: %v", pErr, err)
-		}
-		return errors.Wrap(err, "exec benchmark A")
+		return nil, errors.Wrap(err, "not clean worktree")
 	}
 
-	var oldResult string
-
-	targetCommit, compareWithItself, err := compareTargetRef(ctx, env.Repo(), env.CompareTarget())
+	// Get info about target.
+	targetCommit, compareWithItself, err := getTargetInfo(ctx, env.Repo(), env.CompareTarget())
 	if err != nil {
-		return errors.Wrap(err, "compareTargetRef")
+		return nil, errors.Wrap(err, "getTargetInfo")
 	}
+	bench.logger.Println("Target:", targetCommit.String(), "Current Ref:", ref.Hash().String())
 
 	if compareWithItself {
-		bench.logger.Println("Target:", env.CompareTarget(), "is `.`; Assuming sub-benchmarks comparison.")
+		bench.logger.Println("Assuming sub-benchmarks comparison.")
+		subResult, err := bench.execBenchmark(wt.Filesystem.Root(), ref.Hash())
+		if err != nil {
+			return nil, errors.Wrap(err, "execute sub-benchmark")
+		}
 
-		// 2a. Compare sub benchmarks. TODO.
-		cmps, err := bench.compareSubBenchmarks(newResult)
+		cmps, err := bench.compareSubBenchmarks(subResult)
 		if err != nil {
-			if pErr := env.PostErr("`benchcmp` failed."); pErr != nil {
-				return errors.Errorf("error: %v occured while processing error: %v", pErr, err)
-			}
-			return errors.Wrap(err, "compare sub benchmarks")
+			return nil, errors.Wrap(err, "comparing sub benchmarks")
 		}
-		return errors.Wrap(env.PostResults(cmps), "post results")
+		return cmps, nil
 	}
 
-	bench.logger.Println("Target:", env.CompareTarget(), "is evaluated to be ", targetCommit.String(), ". Assuming comparing with this one (clean workdir will be checked.)")
-	// 2b. Compare with target commit/branch.
+	bench.logger.Println("Assuming comparing with target (clean workdir will be checked.)")
 
-	// 3. Best effort cleanup of worktree.
-	cmpWorkTreeDir := filepath.Join(wt.Filesystem.Root(), "_funcbench-cmp")
+	// Execute benchmark A.
+	newResult, err := bench.execBenchmark(wt.Filesystem.Root(), ref.Hash())
+	if err != nil {
+		return nil, errors.Wrapf(err, "execute benchmark for A: %v", ref.Name().String())
+	}
 
-	_, _ = bench.c.exec("git", "worktree", "remove", cmpWorkTreeDir)
-	bench.logger.Println("Checking out (in new workdir):", cmpWorkTreeDir, "commmit", ref.String())
-	if _, err := bench.c.exec("git", "worktree", "add", "-f", cmpWorkTreeDir, ref.Hash().String()); err != nil {
-		return errors.Wrapf(err, "failed to checkout %s in worktree %s", ref.String(), cmpWorkTreeDir)
+	// Best effort cleanup and checkout new worktree.
+	if err := os.RemoveAll(cmpWorkTreeDir); err != nil {
+		return nil, errors.Wrapf(err, "delete worktree at %s", cmpWorkTreeDir)
 	}
 
-	// 4. Benchmark in new worktree.
-	oldResult, err = bench.execBenchmark(cmpWorkTreeDir, targetCommit)
+	if _, err := bench.c.exec("git", "worktree", "prune"); err != nil {
```

Comment:
I don't think this is nice. What you have locally 10000 important worktrees? We should remove just one

Reply:
I added this because @nevill mentioned that it's not working with git < 2.17, otherwise we can mention that funcbench requires git 2.17+  ?

But i tested this, `prune` seems to remove worktrees that no longer exists so it will not remove other worktrees. I'll test it again tomorrow just to confirm.

Reply:
For `prune`, we have to `rm` the directory first, then prune.

In fact, I prefer `go-git` to do it, in that case we can remove `git` CLI as dependency in container image. Maybe another pull request to fix it.

Reply:
As long as we use git >= 2.17, it's fine. I checked current alpine based image, it's v2.24.1.

Reply:
go git has zero support of worktrees

Reply:
It's not the best library :/  

Anyway, everything is better than malforming git repo for someone! (:

Reply:
If prune works on not used ones, then fine I guess (: